boards/nrf51dk/Makefile

```diff
@@ -1,26 +1,35 @@
 # Makefile for building the tock kernel for the nRF development kit
 
+SIZE?=arm-none-eabi-size
 OBJCOPY?=arm-none-eabi-objcopy
 OBJDUMP?=arm-none-eabi-objdump
 OBJDUMP_FLAGS+= --disassemble-all --source --disassembler-options=force-thumb -C --section-headers
 
 .PHONY: all
-all: target/target/release/nrf51dk target/target/release/nrf51dk.lst
+all: target/nrf51/release/nrf51dk
 
 .PHONY: doc
 doc:
-	@cargo doc --release --target=../../chips/nrf51/target.json
+	@cargo doc --release --target=nrf51.json
 
-.PHONY: target/target/release/nrf51dk
-target/target/release/nrf51dk:
-	@cargo build --release --target=../../chips/nrf51/target.json
+.PHONY: target/nrf51/release/nrf51dk
+target/nrf51/release/nrf51dk:
+	@cargo build --release --target=nrf51.json
+	@$(SIZE) $@
 
-target/target/release/nrf51dk.lst: target/target/release/nrf51dk
-	@$(OBJDUMP) $(OBJDUMP_FLAGS) $^ > $@
+.PHONY: target/nrf51/debug/nrf51dk
+target/nrf51/debug/nrf51dk:
+	@echo $(RUSTFLAGS)
+	@cargo build --target=nrf51.json
+	@$(OBJDUMP) $(OBJDUMP_FLAGS) $@ > target/nrf51/debug/nrf51dk.lst
```

Comment:
@alevy I modified the kernel Makefiles to have a `make debug` option that builds an unoptimzed binary, will in the future build with debug symbols (once that isn't broken), and generates an LST file.

I tried, but could _not_ figure out a way to generate the Map file in debug only. I can't seem to:
 * pass linker arguments to `cargo build`
 * add linker arguments to `Cargo.toml` for the `dev` profile only
 * generate a Map file after `cargo build` is complete

Any ideas? It's not vital to generate a Map file, but they are useful when debugging.